server/utils/utils.ts

```diff
@@ -74,83 +74,80 @@ export const formatAssessmentData = (
-          ?.value
-      const strengthsOrProtectiveFactors =
-        assessment.sanAssessmentData[
-          `${area.assessmentKey}_practitioner_analysis_strengths_or_protective_factors_yes_details`
-        ]?.value
-
-      return {
-        title: area.area,
-        overallScore: overallScore ?? score,
-        linkedtoRoSH,
-        linkedtoReoffending,
-        motivationToMakeChanges,
-        riskOfSeriousHarm,
-        riskOfReoffending,
-        strengthsOrProtectiveFactors,
-        criminogenicNeedsScore: score,
-        goalRoute: area.goalRoute,
-        upperBound: area.upperBound,
-        thresholdValue: getAssessmentAreaThreshold(area.crimNeedsKey),
-        subData,
-      } as AssessmentArea
-    })
-    .sort((a, b) =>
-      Number(b.criminogenicNeedsScore) - b.thresholdValue - (Number(a.criminogenicNeedsScore) - a.thresholdValue) > 0
-        ? 1
-        : -1,
-    )
-
-  const lowScoring = all.filter(area => Number(area.overallScore) <= area.thresholdValue)
-  const highScoring = all.filter(area => Number(area.overallScore) > area.thresholdValue)
+    }
+
+    const motivationToMakeChanges = motivationText(assessment.sanAssessmentData[`${area.assessmentKey}_changes`]?.value)
+    const riskOfSeriousHarm =
+      assessment.sanAssessmentData[`${area.assessmentKey}_practitioner_analysis_risk_of_serious_harm_yes_details`]
+        ?.value
+    const riskOfReoffending =
+      assessment.sanAssessmentData[`${area.assessmentKey}_practitioner_analysis_risk_of_reoffending_yes_details`]?.value
+    const strengthsOrProtectiveFactors =
+      assessment.sanAssessmentData[
+        `${area.assessmentKey}_practitioner_analysis_strengths_or_protective_factors_yes_details`
+      ]?.value
+
+    return {
+      title: area.area,
+      overallScore: overallScore ?? score,
+      linkedtoRoSH,
+      linkedtoReoffending,
+      motivationToMakeChanges,
+      riskOfSeriousHarm,
+      riskOfReoffending,
+      strengthsOrProtectiveFactors,
+      criminogenicNeedsScore: score,
+      goalRoute: area.goalRoute,
+      upperBound: area.upperBound,
+      thresholdValue: getAssessmentAreaThreshold(area.crimNeedsKey),
+      subData,
+    } as AssessmentArea
+  })
+
+  const lowScoring = filterAndSortAreas(all, (score, threshold) => score <= threshold)
+  const highScoring = filterAndSortAreas(all, (score, threshold) => score > threshold)
   const other = all.filter(area => area.criminogenicNeedsScore === undefined)
   return { lowScoring, highScoring, other, versionUpdatedAt: assessment.lastUpdatedTimestampSAN } as AssessmentAreas
 }
 
+const filterAndSortAreas = (areas: AssessmentArea[], comparator: (score: number, threshold: number) => boolean) => {
+  return areas
+    .filter(area => comparator(Number(area.overallScore), area.thresholdValue))
+    .sort((a, b) => (Number(a.overallScore) - a.thresholdValue > Number(b.overallScore) - b.thresholdValue ? -1 : 1))
+}
```

Comment:
Because this always returns -1 or 1, will this cause odd shifting in the sort order when the values of `a` and `b` are the same?

Reply:
Yes. I'll add an extra check for score equality and sort by title.

Reply:
Fixed in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/260/commits/6be11af672980c9e4990cddececcf804bfd7efd8